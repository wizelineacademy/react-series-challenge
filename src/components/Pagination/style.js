import styled from 'styled-components'

export const PaginationButton = styled.span`
  color: black;
  background-color: white;
  padding: 8px 16px;
  text-decoration: none;
  transition: background-color .3s;
  border: 1px solid #ddd;
  cursor: pointer;
  margin-top: 30px;
  ${ props => props.active && 
    `background-color: #4CAF50;
    color: white;
    border: 1px solid #4CAF50;
    cursor: default;
    `
  }
`