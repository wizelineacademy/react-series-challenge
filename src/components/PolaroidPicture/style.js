import styled from 'styled-components'
import { Link } from 'react-router-dom';


export const PolaroidListItem = styled.li`
  position: relative;
  width: ${props => props.isSingleView ? '500px;' : '200px;'}
  height: ${props => props.isSingleView ? '650px;' : '250px;'}
  display: inline-block;
  padding: 10px;
  margin: 10px;
  background-color: white;
  box-shadow: 7px 7px 4px black;
  border: 1px solid #000
  img {
    max-width: 100%;
    height: 80%;
  }
`
export const PolaroidLink = styled(Link)`
  color: white;
`

export const PolaroidStar = styled.div`
  background-image: url('/favorited.ico');
  background-size: contain;
  width: 50px;
  height: 50px;
  position: absolute;
  top: -25px;
  right: calc(50% - 25px);
`

export const PolaroidButton = styled.button`
  margin: 5px;
  padding: 10px 19px;
  margin: 0px 10px -10px 0px;
  border-radius: 10px;
  font-family: 'Pacifico', cursive;
  font-size: 17px;
  color: #FFF;
  text-decoration: none; 
  border: none;
  outline:none;
  cursor: pointer;
  display: none;
  ${ props => props.color === 'blue' ?
      `
        background-color: #3498DB;
        text-shadow: 0px -2px #2980B9;
      `
      :
      `
        background-color: #82BF56;
        text-shadow: 0px -2px #669644;
        float: right;
      `
  }
  :active {
    transform: translate(0px,5px);
    -webkit-transform: translate(0px,5px);
    border-bottom: 1px solid;
  }
`

export const PolaroidLoading = styled.span`
  position: absolute;
  width: 200px;
  height: 200px;
  left: 10px;
  background-image: url("/loader.svg");
  background-size: contain;
  background-color: black;
`