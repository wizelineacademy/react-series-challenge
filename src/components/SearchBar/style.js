import styled from 'styled-components'

export const Container = styled.div `
  display: flex;
  justify-content: center;
  margin-bottom: 70px;
`

export const SearchBox = styled.div `
  position: absolute;
  input {
    border: none;
    background: none;
    z-index: 1;
    width: 50px;
    height: 50px;
    transition: all .25s ease-in .25s;
    color: transparent;
    font-size: 1.25rem;
    line-height:50px;
    &:hover {
      cursor: pointer;
      &:focus {
        cursor: text;
      }
      + span {
        background: rgba(255,255,255,0.2);
      }
    }
    &:focus {
      width: 200px;
      padding: 0 10px;
      outline: none;
      color: black;
      background:none;
      color: white;
      + span {
        width: 200px;
        &::before {
          width:2px;
          opacity: 0;
          transition: all .25s ease-in;
        }
      }
    }
    + span {
      z-index: -1;
      position: absolute;
      border: 2px solid white;
      top: 0;
      width: 50px;
      height: 50px;
      transition: all .25s ease-in .25s;
      border-radius: 50px;
      left: 0;
      &::before {
        transition: all .25s ease-in .5s;
        transform-origin: left top;
        content: '';
        position: absolute;
        width: 20px;
        height: 7px;
        border-radius: 5px;
        background: white;
        transform: rotate(45deg) translate(43px, 10px);
      }
    }
  }
`