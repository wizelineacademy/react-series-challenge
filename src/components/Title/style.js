import styled from 'styled-components'

export const BurningEffect = styled.h1`
  color: #fff;
  text-shadow: 0px -1px 4px white, 0px -2px 10px yellow, 0px -10px 20px #ff8000, 0px -18px 40px red;
  font: 2em 'BlackJackRegular';
`

export const NeonEffect = styled.h1`
  color: #fff;
  text-shadow: 0 0 5px #fff, 0 0 10px #fff, 0 0 15px #fff, 0 0 20px #ff2d95, 0 0 30px #ff2d95, 0 0 40px #ff2d95, 0 0 50px #ff2d95, 0 0 75px #ff2d95;
  letter-spacing: 5px;
  font: 3em 'Miso';
`