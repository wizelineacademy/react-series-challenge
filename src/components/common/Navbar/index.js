import React from 'react'
import NavbarStyled from './NavbarStyled'
import Link from '../Link'



const Navbar = (props)=>{
    return(
        <NavbarStyled>
                <Link to="/">Home</Link>
                <Link to="/favorites">Favorites</Link>
        </NavbarStyled>
    )
}

export default Navbar