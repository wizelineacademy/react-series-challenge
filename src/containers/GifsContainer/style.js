import styled from 'styled-components'
import { NavLink } from 'react-router-dom';

export const PolaroidContainer = styled.ul`
  ${props => props.singleView ? 'left: calc(50% - 300px);' : ''}

  li:nth-child(even) {
    transform: rotate(2deg);
    -webkit-transform: rotate(2deg);
    -moz-transform: rotate(2deg);
  }
  li:nth-child(3n) {
    transform: rotate(none);
    -webkit-transform: none;
    -moz-transform: none;
    position: relative;
    top: -5px;
  }
  li:nth-child(5n) {
    transform: rotate(5deg);
    -webkit-transform: rotate(5deg);
    -moz-transform: rotate(5deg);
    position: relative;
    right: 5px;
  }
  li:nth-child(7n) {
    transform: rotate(-3deg);
    -webkit-transform: rotate(-3deg);
    -moz-transform: rotate(-3deg);
    position: relative;
    right: -5px;
  }
  li:nth-child(8n) {
    position: relative;
    top: 8px;
    right: 5px;
  }
  li:nth-child(11n) {
    position: relative;
    top: 3px;
    left: -5px;
  }
  ${props => !props.isSingleView && `
    li:hover {
      transform: scale(1.25);
      -webkit-transform: scale(1.25);
      -moz-transform: scale(1.25);
      transition: all 0.2s ease-out;
      position: relative;
      z-index: 5;
      display: inline-table;
      button {
        display: inline-block;
      }
    }
  `}
`
export const SquareLink = styled(NavLink)`
  background-color: #00c08b;
  color: #127a5d;
  height: 50px;
  width: 50px;
  box-shadow: 0 0 10px rgba(100, 100, 100, 0.2) inset, 1px 1px 1px #CCC;
  display: inline-block;
  line-height: 1;
  margin: 5px;
  border-radius: 5px;
  position: relative;
  background-image: url('/home.svg');
  ${ props => props.favorites &&
    `background-image: url('/star.svg');
    background-color: #53bfe2;`
  }
  background-size: contain;
  &.active {
    display: none;
  }
`
