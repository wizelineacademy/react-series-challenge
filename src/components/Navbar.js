import React, { Component } from "react";
import { NavLink, Link } from "react-router-dom";
import {UlStyled, LiStyled, InputTextStyled,ButtonTextStyled} from "./Navbar.style"

class Navbar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tag: "nothing"
    };
  }
  render() {
    return (
      <React.Fragment>
        <UlStyled>
          <LiStyled>
            <NavLink to="/">Home</NavLink>
          </LiStyled>
          <LiStyled>
          <NavLink to="/fav">Favs</NavLink>
          </LiStyled>
          <LiStyled>
          <InputTextStyled
              type="text"
              name="search"
              onChange={event => { this.setState({ tag: event.target.value }); }}
            />
          </LiStyled>
          <LiStyled>
            <Link
              to={{
                pathname: "/search",
                state: { query: this.state.tag }
              }}
            >
              <ButtonTextStyled type="button" id="add">Search</ButtonTextStyled>
            </Link>
          </LiStyled>
        </UlStyled>
      </React.Fragment>
    );
  }
}

export default Navbar;