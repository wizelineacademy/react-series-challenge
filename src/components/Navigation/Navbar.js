import React from "react";
import { withRouter } from "react-router-dom"
import SearchBar from "../SearchBar/SearchBar";
import { ContainerNavbarStyled, NavbarLinkStyle } from "./NavbarStyle";

const Navbar = ({ location }) => (
  // location.pathame
  <ContainerNavbarStyled>
    <div>
      <NavbarLinkStyle exact to="/">Home</NavbarLinkStyle>
      <NavbarLinkStyle to="/favorites">Favoritos</NavbarLinkStyle>
    </div>
    <SearchBar from={location.pathname}/>
  </ContainerNavbarStyled>
);

export default withRouter(Navbar);
