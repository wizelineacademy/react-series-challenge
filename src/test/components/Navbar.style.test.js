import React from 'react'
import renderer from 'react-test-renderer'
import { ButtonTextStyled } from "../../components/Navbar.style"

test('Styled navbar works', () => {
  const tree = renderer.create(<ButtonTextStyled />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('color', 'white')
})