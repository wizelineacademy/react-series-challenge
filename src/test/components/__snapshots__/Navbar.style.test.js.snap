// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled gif works 1`] = `
<button
  className="sc-bxivhb jhBOgb"
/>
`;

exports[`Styled navbar works 1`] = `
<button
  className="sc-bxivhb jhBOgb"
/>
`;
